ospfd/ospf_interface.c

```diff
@@ -1447,6 +1447,46 @@ static int ospf_ifp_destroy(struct interface *ifp)
 	return 0;
 }
 
+/* Resetting ospf hello timer */
+void ospf_reset_hello_timer(struct interface *ifp, struct in_addr addr,
+			    bool is_addr)
+{
+	struct route_node *rn;
+
+	if (is_addr) {
+		struct prefix p;
+		struct ospf_interface *oi = NULL;
+
+		p.u.prefix4 = addr;
+		p.family = AF_INET;
+
+		oi = ospf_if_table_lookup(ifp, &p);
+
+		if (oi) {
+			/* Restart hello timer for this interface */
+			OSPF_ISM_TIMER_OFF(oi->t_hello);
+			OSPF_HELLO_TIMER_ON(oi);
+		}
+
+		return;
+	}
+
+	for (rn = route_top(IF_OIFS(ifp)); rn; rn = route_next(rn)) {
+		struct ospf_interface *oi = rn->info;
+
+		if (!oi)
+			continue;
+
+		/* If hello interval configured on this oi, don't restart. */
+		if (OSPF_IF_PARAM_CONFIGURED(oi->params, v_hello))
+			continue;
+
+		/* Restart the hello timer. */
+		OSPF_ISM_TIMER_OFF(oi->t_hello);
+		OSPF_HELLO_TIMER_ON(oi);
```

Comment:
nit: maybe it's worth sending an extra hello before restarting this timer. That would protect against session flaps if the new interval is too big, or if this command is entered repeatedly in short spaced intervals.

Reply:
This can become a real issue if the neighbor's dead interval is not big enough, so I think it should be fixed.

Reply:
same problem on lines 1467 and 1468

Reply:
Addressed the suggested change.

Reply:
I really don't see how anything has been changed here.  What am I missing?

Reply:
@donaldsharp ,  
Called ospf_hello_send()  api  to send hello once to the peer  before restarting the hello timer.